projects/igniteui-angular/src/lib/grid/grid-navigation.service.ts

```diff
@@ -62,12 +67,48 @@ export class IgxGridNavigationService {
             this.grid.dataRowList.first.virtDirRow.getColumnScrollLeft(index);
     }
 
+    public get gridOrderedColumns(): IgxColumnComponent[] {
+        return [...this.grid.pinnedColumns, ...this.grid.unpinnedColumns].filter(c => !c.columnGroup);
+    }
+
+    public isRowInEditMode(rowIndex): boolean {
+        return this.grid.rowEditable && (this.grid.rowInEditMode && this.grid.rowInEditMode.index === rowIndex);
+    }
+
+    public isColumnEditable(visibleColumnIndex: number): boolean {
+        const column = this.gridOrderedColumns.find(c => c.visibleIndex === visibleColumnIndex);
+        return column ? column.editable : false;
+    }
+
+    public findNextEditable(direction: string, visibleColumnIndex: number) {
+        const gridColumns = this.gridOrderedColumns;
+        if (direction === MoveDirection.LEFT) {
+            return gridColumns.splice(0, visibleColumnIndex + 1).reverse().findIndex(e => e.editable);
+        } else if (direction === MoveDirection.RIGHT) {
+            return gridColumns.splice(visibleColumnIndex, gridColumns.length - 1).findIndex(e => e.editable);
+        }
+    }
+
     public isColumnLeftPartiallyVisible(visibleColumnIndex) {
         const index = this.getColumnUnpinnedIndex(visibleColumnIndex + 1);
         return (!this.isColumnLeftFullyVisible(visibleColumnIndex)) && this.displayContainerScrollLeft <
             this.grid.dataRowList.first.virtDirRow.getColumnScrollLeft(index);
     }
 
+    public getCellElementByVisibleIndex(rowIndex, visibleColumnIndex) {
+        return this.grid.nativeElement.querySelector(
+            `igx-grid-cell[data-rowindex="${rowIndex}"][data-visibleIndex="${visibleColumnIndex}"]`) || { focus: () => {}};
```

Comment:
This was implemented instead of a null check on each of the places where the `getCellElementByVisibleIndex` method was called - due to performance or the browser handling the `tab` keydown, you'd get the focus "fly out" of the grid when holding down tab and it would result in errors. With the improvement to the tab handling (with the listener in the grid component) that should not be an issue - should I remove it?

Reply:
Yes, it should not be needed anymore.